src/main.rs

```diff
@@ -255,6 +301,35 @@ fn main() {
             exit(-6);
         }
     }
+
+    // Clean up temporary directory if we created one
+    if args.use_temp_dir {
+        info!("Cleaning up temporary directory on remote server...");
+        let cleanup_result = Command::new("ssh")
+            .args(&["-p", &remote.ssh_port.to_string()])
+            .arg(&build_server)
+            .arg(format!("rm -rf '{}'", build_path))
```

Comment:
I would be curious if `cargo clean` might be nice to run before `rm`, so the user gets a progress indicator when deleting a large `target` dir.

Also would prefer to use `rm -r` rather than `-rf` if that also works.